composer/callbacks/early_stopper.py

```diff
@@ -0,0 +1,139 @@
+        comp (Callable[[Any, Any], bool], optional): A comparison operator to measure change of the monitored metric. The comparison
+            operator will be called `comp(current_value, prev_best)`. For metrics where the optimal value is low
+            (error, loss, perplexity), use a less than operator and for metrics like accuracy where the optimal value
+            is higher, use a greater than operator. Defaults to numpy.less if loss, error, or perplexity are substrings
+            of the monitored metric, otherwise defaults to numpy.greater
+        min_delta (float, optional): An optional float that requires a new value to exceed the best value by at
+            least that amount. Defaults to 0.
+        patience (int | str | Time, optional): The interval of time the monitored metric can not improve without stopping
+            training. Defaults to 1 epoch.
+        
+    """
+
+    def __init__(
+        self,
+        monitor: str,
+        dataloader_label: str,
+        comp: Optional[Callable[[
+            Any,
+            Any,
+        ], bool]] = None,
+        min_delta: float = 0.0,
+        patience: Union[int, str, Time] = 1,
+    ):
+        self.monitor = monitor
+        self.dataloader_label = dataloader_label
+        self.comp = comp
+        self.min_delta = abs(min_delta)
+        if self.comp is None:
+            if any(substr in monitor.lower() for substr in ["loss", "error", "perplexity"]):
+                self.comp = np.less
+            else:
+                self.comp = np.greater
+
+        self.best = None
+        self.best_occurred = None
+
+        if isinstance(patience, str):
+            self.patience = Time.from_timestring(patience)
+        elif isinstance(patience, int):
+            self.patience = Time(patience, TimeUnit.EPOCH)
+        else:
+            self.patience = patience
+            if self.patience.unit not in (TimeUnit.EPOCH, TimeUnit.BATCH):
+                raise ValueError("If `patience` is an instance of Time, it must have units of EPOCH or BATCH.")
+
+    def _get_monitored_metric(self, state: State):
+        if self.dataloader_label in state.current_metrics:
+            if self.monitor in state.current_metrics[self.dataloader_label]:
+                return state.current_metrics[self.dataloader_label][self.monitor]
+        raise ValueError(f"Couldn't find the metric {self.monitor} with the dataloader label {self.dataloader_label}."
+                         "Check that the dataloader_label is set to 'eval', 'train' or the evaluator name.")
+
+    def _update_stopper_state(self, state: State):
+        metric_val = self._get_monitored_metric(state)
+
+        assert self.comp is not None
+        if self.best is None:
+            self.best = metric_val
+            self.best_occurred = state.timer.get_timestamp()
+        elif self.comp(metric_val, self.best) and abs(metric_val - self.best) > self.min_delta:
```

Comment:
Does this need to be torch.abs? See comment re: gpu

```suggestion
        elif self.comp(metric_val, self.best) and torch.abs(metric_val - self.best) > self.min_delta:
```